services/storage/src/simcore_service_storage/db_file_meta_data.py

```diff
@@ -119,7 +119,7 @@ async def list_filter_with_partial_file_id(
     file_id_prefix: str | None,
     partial_file_id: str | None,
     sha256_checksum: SHA256Str | None,
-    only_files: bool,
+    is_directory: bool | None = None,
```

Comment:
even here, this is a public function but you control all the callers.
I would suggest to keep it optional, since you might want to use the filter or not. But I would still make it a required argument, there is no backwards compatibility issue here.